.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,116 @@
+name: Docker Build, Publish and Deploy
+
+on:
+  push:
+    branches: [master]
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  docker_build_and_publish:
+    runs-on: ubuntu-latest
+    outputs:
+      IMAGE_TAGE: ${{ steps.image_tag.outputs.IMAGE_TAG }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Define_docker_image_tag
+        run: |
+          echo "DOCKER_IMAGE_TAG=(git describe --tags)" >> $GITHUB_ENV
+          echo "IMAGE_TAG=$(git describe --tags)" >> "$GITHUB_OUTPUT"
+      
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with: 
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Build and Push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: 'linux/amd64'
+          push: true
+          tags: nethermindeth/juno:${{ github.event.inputs.tag }}
+    
+  deploy_to_dev:
+    permissions:
+      id-token: write
+      contents: write
+    needs: [docker_build_and_publish]
+    runs-on: ubuntu-latest
+    environment: 
+      name: Development
+    steps:
+      - name: Repository Dispatch Dev
+        env:
+          EVENT_NAME: juno-dev
+          IMAGE_TAG: ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
+          GOERLI: apps/juno-dev/overlays/dev-goerli-1/config.yaml
+          GOERLI2: apps/juno-dev/overlays/dev-goerli-2/config.yaml
+          TESTNET: apps/juno-dev/overlays/dev-integration/config.yaml
+          MAINNET: apps/juno-dev/overlays/dev-mainnet/config.yaml
+        run: |
+          curl -L \
+          -X POST \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: token ${{ secrets.ACCESS_TOKEN }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          https://api.github.com/repos/NethermindEth/argo/dispatches \
+          -d '{"event_type": "${{ env.EVENT_NAME }}", "client_payload":{"name": "${{ env.EVENT_NAME }}", "goerli_config": "${{ env.GOERLI }}", "goerli2_config": "${{ env.GOERLI2 }}", "testnet_config": "${{ env.TESTNET }}", "mainnet_config": "${{ env.MAINNET }}", "tag": "${{ env.IMAGE_TAG }}"}}'
+
+  deploy_to_staging:
+    needs: [docker_build_and_publish, deploy_to_dev]
+    runs-on: ubuntu-latest
+    environment: 
+      name: Staging
+    steps:
+      - name: Repository Dispatch Staging
+        env:
+          EVENT_NAME: juno-staging
+          IMAGE_TAG: ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
+          GOERLI: apps/juno-staging/overlays/staging-goerli-1/config.yaml
+          GOERLI2: apps/juno-staging/overlays/staging-goerli-2/config.yaml
+          TESTNET: apps/juno-staging/overlays/staging-integration/config.yaml
```

Comment:
Can we name INTEGRATION network as INTEGRATION, not as TESTNET? I mean Goerli is also a testnet so it can be misleading.